sql: log AST and plan gist for most sentry reports

DrewKimball opened the pull request:
#### sql: don't redact post-query type from errors

This commit fixes a minor oversight from when we introduced AFTER
triggers: the type of post-query (cascade vs trigger) is redacted in
error messages. This information is safe, so this commit marks the
action type as a `SafeString`.

Epic: None

Release note: None

#### sql: remove unnecessary param from helper functions

This commit removes an unused `ClientNoticeSender` parameter from
several functions related to formatting virtual table names.

Epic: None

Release note: None

#### sql: log AST and plan gist for most sentry reports

This commit adds a method to the `connExecutor` that annotates an
error with the currently-executing statement's AST and plan gist, if
available (saved on the `connExecutor`). This method is now called
before an error is used to build a sentry report in most cases
(i.e. any time an error or panic is propagated up to the `connExecutor`).

Fixes #93285

Release note: None

DrewKimball committed:
sql: don't redact post-query type from errors

This commit fixes a minor oversight from when we introduced AFTER
triggers: the type of post-query (cascade vs trigger) is redacted in
error messages. This information is safe, so this commit marks the
action type as a `SafeString`.

Epic: None

Release note: None

yuzefovich reviewed (approved):
Nice! This won't solve the missing stmt / gist issue in all callsites, but I agree that this should capture the most common one. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @yuzefovich reviewed 1 of 1 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152367#-:-OYJEKuW-dFhQXDrJd2L:b-hi9bqk)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)

___
*[`pkg/sql/exec_util.go` line 2795 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/152367#-OYJBAXi11yGPuIz3rdr:-OYJBAXi11yGPuIz3rds:b-x6ioxw) ([raw file](https://github.com/cockroachdb/cockroach/blob/6af5d71d891ee5b45a607ce8d09b4ca10acb72fd/pkg/sql/exec_util.go#L2795)):*
> ```Go
> 		f = tree.NewFmtCtx(
> 			fmtFlags,
> 			tree.FmtReformatTableNames(hideNonVirtualTableNameFunc(vt, nil)),
> ```

nit: rather than passing `nil` as an argument, we should just remove that argument altogether given that it's not actually used in the implementation of `VirtualTabler.getVirtualTableEntry`. Seems worthy of extracting that change into a separate commit too.


<!-- Sent from Reviewable.io -->

DrewKimball committed:
sql: remove unnecessary param from helper functions

This commit removes an unused `ClientNoticeSender` parameter from
several functions related to formatting virtual table names.

Epic: None

Release note: None

DrewKimball committed:
sql: log AST and plan gist for most sentry reports

This commit adds a method to the `connExecutor` that annotates an
error with the currently-executing statement's AST and plan gist, if
available (saved on the `connExecutor`). This method is now called
before an error is used to build a sentry report in most cases
(i.e. any time an error or panic is propagated up to the `connExecutor`).

Fixes #93285

Release note: None

DrewKimball reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152367)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)

___
*[`pkg/sql/exec_util.go` line 2795 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/152367#-OYJBAXi11yGPuIz3rdr:-OYmgUwo904krcSv2u0Z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6af5d71d891ee5b45a607ce8d09b4ca10acb72fd/pkg/sql/exec_util.go#L2795)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: rather than passing `nil` as an argument, we should just remove that argument altogether given that it's not actually used in the implementation of `VirtualTabler.getVirtualTableEntry`. Seems worthy of extracting that change into a separate commit too.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @mgartner reviewed 1 of 1 files at r1, 5 of 5 files at r3, 9 of 9 files at r4, 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152367#-:-OZPWiMAC37G1QWiixXv:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @aa-joshi, @Abhinav1299, @alyshanjahani-crl, and @arjunmahishi)



<!-- Sent from Reviewable.io -->

DrewKimball commented:
TFTRs!

bors r+
